compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -629,9 +629,8 @@ impl<'f, 'tcx> Coerce<'f, 'tcx> {
         while !queue.is_empty() {
             let obligation = queue.remove(0);
             debug!("coerce_unsized resolve step: {:?}", obligation);
-            let bound_predicate = obligation.predicate.kind();
-            let trait_pred = match bound_predicate.skip_binder() {
-                ty::PredicateKind::Clause(ty::ClauseKind::Trait(trait_pred))
+            let trait_pred = match obligation.predicate.kind().no_bound_vars() {
```

Comment:
This *could* technically affect inference, but `CoerceUnsized` is an unstable trait and I don't know of a stdlib impl that emits bound predicates :^)